### PR TITLE
Hide actions that start with _ and have flag to show them

### LINKED ISSUE #301: q2cli should hide actions with names that start with `_`
- `_` shouldn't be replaced with `-`
- actions starting with underscores should be hidden by default, with a flag to include them in the list when called as `qiime plugin-name`

### PR DESCRIPTION
**Pull request by Oddant1**
Closes #301. Hides actions that start with '_' by default and adds a flag to display them.

~~For click related reasons that I do not fully understand yet, this does not work and passing the command results in `Error: Missing Command`  think I more or less understand why it's happening, but I don't have a good solution for the issue yet.~~

Should be gtg now.

### CODE REVIEW

**Comment by Oddant1** [comment]
~~blocked by qiime2/qiime2#710~~

**Comment by Oddant1** [comment]
~~This PR does not currently allow you to actually execute the hidden method from the command line. Need to fix that~~

File: q2cli/util.py [hunk 1]
```diff
@@ -36,6 +36,15 @@ def get_completion_path():
     return os.path.join(get_cache_dir(), 'completion.sh')
 
 
+def hidden_to_cli_name(name):
+    # Safety first
+    assert name.startswith('_')
```
Comment (gregcaporaso) [inline q2cli/util.py]: I'm assuming you always want this to raise an error if the name doesn't start with `_` - is that right? If so, it's better to use if/raise as [`assert` statements aren't always evaluated](https://stackoverflow.com/a/1838411/3424666).

File: q2cli/commands.py [hunk 2]
```diff
@@ -171,6 +189,28 @@ def _get_citation_records(self):
...
+        # which for this is False. Honestly not 100% sure why we need that
+        # here, but it is in the check in _get_version, and I a mimicking that
+        if not value or ctx.resilient_parsing:
+            return
+
+        from click.utils import echo
+
+        self._action_lookup.update(self._hidden_actions)
+
+        # Handle the printing and exiting here. This feels like a pretty
```
Comment (gregcaporaso) [inline q2cli/commands.py:208]: I'm not familiar enough with click to know if there is a better way. Maybe worth getting some quick input from @ebolyen on this one.

**Review by gregcaporaso — COMMENTED** [review]
In general this looks good to me - couple of comments.

File: q2cli/commands.py (hunk 2 above)
Comment (ebolyen) [reply q2cli/commands.py:208]: Nope, this is exactly the right API it seems.

**Review by ebolyen — APPROVED** [review]
Looks great, thanks @Oddant1!

**Comment by ebolyen** [comment]
Tests pass locally with the latest framework, so merging now.

**Pull request merged**